pkg/rancher/migrations.go

```diff
@@ -736,6 +746,51 @@ func migrateHarvesterCloudCredentialExpiration(w *wrangler.Context) error {
 	return createOrUpdateConfigMap(w.Core.ConfigMap(), cm)
 }
 
+// migrateImportedClusterFields will update fields on clusters.management.cattle.io/v3 for imported clusters,
+// fields will no longer be set by clusters.provisioning.cattle.io/v1 ensuring management cluster
+// remains the source of truth for imported clusters.
+func migrateImportedClusterFields(w *wrangler.Context) error {
+	cm, err := getConfigMap(w.Core.ConfigMap(), migrateImportedClusterManagedFields)
+	if err != nil || cm == nil {
+		return err
+	}
+
+	if cm.Data[importedClusterManagedFieldsMigratedKey] == "true" {
+		return nil
+	}
+
+	provClusters, err := w.Provisioning.Cluster().List("", metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, cluster := range provClusters.Items {
+		if mgmtNameRegexp.MatchString(cluster.Name) || cluster.Spec.RKEConfig != nil {
+			// run only for non-legacy imported clusters
+			continue
+		}
+		mgmtCluster, err := w.Mgmt.Cluster().Get(cluster.Status.ClusterName, metav1.GetOptions{})
+		if k8serror.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return err
+		}
+		clusterCopy := mgmtCluster.DeepCopy()
+		// clusterDeploy controller uses GetDesiredAgentImage(mgmtCluster), customizable using Spec.DesiredAgentImage
+		clusterCopy.Spec.DesiredAgentImage = ""
+		// clusterDeploy controller uses GetDesiredAuthImage(mgmtCluster), customizable using Spec.DesiredAuthImage
+		clusterCopy.Spec.DesiredAuthImage = ""
+		// managed by Spec.ImportedConfig.PrivateRegistryURL for imported clusters
+		clusterCopy.Spec.ClusterSecrets.PrivateRegistryURL = ""
```

Comment:
Could you explain a bit more why we need to reset this field? Is it for handling the case where the global private registry is changed after upgrading Rancher?

Reply:
This should be used only for management clusters created for rke2/k3s provisioned clusters, not imported clusters. For imported clusters we should use `Spec.ImportedConfig.PrivateRegistryURL` so cleaned it up to avoid confusion.